### PR TITLE
Remove socket_helper.h after NetSocket refactor

### PR DESCRIPTION
**Pull request by Faless**
It should have been removed during the `NetSocket` refactor.
Only the `websocket` module was still using it, it's now using `NetSocketPosix` instead.

### CODE REVIEW

**Comment by akien-mga** [comment]
Android build fails it seems: https://travis-ci.org/godotengine/godot/jobs/434300921

**Comment by Faless** [comment]
Thanks, forgot an include, funny that it compiled on all other platforms.
I guess it was included by some other headers.

File: drivers/unix/net_socket_posix.cpp [hunk 1]
```diff
@@ -126,12 +126,12 @@ static size_t _set_addr_storage(struct sockaddr_storage *p_addr, const IP_Addres
 			addr4->sin_addr.s_addr = INADDR_ANY;
 		}
 
-		copymem(&addr4->sin_addr.s_addr, p_ip.get_ipv4(), 16);
+		copymem(&addr4->sin_addr.s_addr, p_ip.get_ipv4(), 4);
```
Comment (Faless) [inline drivers/unix/net_socket_posix.cpp]: I also found this small issue, which was copied over from `socket_helper.h` during the refactoring.
I don't think it deserves to be in its own commit.
While this looks bad, it's most likely harmless, the struct is actually a `sockaddr_storage` which can usually be seen as a union (although it isn't excatly, as per [RFC2553](https://tools.ietf.org/html/rfc2553)) so it won't write out of that var.
Still, this should be fixed :wink:

**Comment by akien-mga** [comment]
Thanks, looks good!

**Pull request merged**